### ISSUE TITLE
VQ3D PnP good poses much lower than reported

### ISSUE THREAD

**Issue by yzhao520** [EGO4D/episodic-memory#6]
Hi,

I'm following the data preparation step for VQ3D to generate camera poses. I fixed a few issues along the way but the amount of good poses I got is much lower than 2% (reported in the paper). I could get around 11 - 25 good poses out of 9000 frames and sometimes it's 0 good poses. In order to find the poses, I have the following questions:

1. I'm assuming the lower number of good poses is due to some errors from previous steps (number of matches, even intrinsic parameters). Is there any way I could check my results before pnp step?

2. Since the paper has reported results with predicted poses. Is it possible to release the computed poses from paper instead of rerunning the code / steps ourselves?

**Comment by vincentcartillier** [comment]
The number of poses (15%) reported in the paper is for both the val and test sets. I recall that for the val set only these numbers get quite low - 2% sounds about right. Did you try to run the remaining part of the pipeline and see if you get the same numbers as in the paper?
1. You can visualize the poses in 3D using the generated ply file - cameras_poses_final.ply. You can also render the egocentric view at the estimated positions using the following script: https://github.com/EGO4D-Consortium/episodic-memory/blob/main/VQ3D/camera_pose_estimation/Visualization/visualize_render_images.py
2. I can certainly release the poses for the val set. I'll circle back here once this is done.

**Comment by yzhao520** [comment]
I haven't tried the remaining part of the pipeline. The program gets killed when running [superglue tracker](https://github.com/EGO4D/episodic-memory/blob/main/VQ3D/camera_pose_estimation/main.sh#L77). 'm guessing it's probably due to my memory/swap memory is not large enough. 

Thanks for pointing out one way to check the camera poses. I would imagine the rendered images with computed poses will not quite be the same as the original frame. If so, how would you suggest to further debugging?

It will be certainly more convenient for me if you could release the camera poses for training and val set!

**Comment by vincentcartillier** [comment]
A PR has been submitted with the camera poses for the val set (I can only share for the val and not for train at the moment). 

I have double checked and the number of camera poses for the val set is closer to 5% (not 2%). Yet when combined with the test set this number gets up to 15%.

Regarding the camera pose debugging, this part is harder since we do not have ground-truth camera poses. I agree that comparing the rendered image from the scan with the original one will only get you so far - The space might have been scanned at a different time when the video was recorded resulting in objects moved places. The scan reconstruction and texture isn't perfect etc.. 
However, this will give you an idea whether the estimated pose make sense or is completely off.

**Comment by yzhao520** [comment]
Thanks for releasing the val pose info! This is definitely helpful in terms of debugging. Just curious, is this released pose from the final step (i.e., poses from step 4: Temporal constraints and final pose estimation or Step 3: PnP resolution in the paper)?

**Comment by vincentcartillier** [comment]
Yes these are from the final step (step 4). The final poses and final good-poses mask are extracted [here](https://github.com/EGO4D/episodic-memory/blob/0dfec57dc65fa008569a319e409f7db5c3e98bf5/VQ3D/camera_pose_estimation/sfm_api_wsuperglue.py#L408) and [here](https://github.com/EGO4D/episodic-memory/blob/0dfec57dc65fa008569a319e409f7db5c3e98bf5/VQ3D/camera_pose_estimation/sfm_api_wsuperglue.py#L448).

**Comment by yzhao520** [comment]
Actually, I have another question that might be related to this camera pose problem. I have tried visualizing 2D bbox from corresponding 2D VQ annotations (from VQ3D queries). Somehow, if I use the clips extracted from VQ2D pages (step 3: Extract clips for validation and test data from videos.), the annotation is correct. But if I use the clips from VQ3D pages (downloaded using cmd: python -m ego4d.cli.cli --output_directory ~/episodic-memory/VQ3D/data/ --dataset clips --universities unict), the annotation is far off and looks like they are not the same frames. So I wonder which clip should we use? Based on the 2D annotations the clips from VQ2D seem right... but I used 3D clips for the following pose computing step.
